cmd/channel/webhook/main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/icinga/icinga-notifications/internal"
+	"github.com/icinga/icinga-notifications/pkg/plugin"
+	"io"
+	"net/http"
+	"text/template"
+)
+
+type Webhook struct {
+	Method              string `json:"method"`
+	URLTemplate         string `json:"url_template"`
+	RequestBodyTemplate string `json:"request_body_template"`
+
+	tmplUrl         *template.Template
+	tmplRequestBody *template.Template
+}
+
+func (ch *Webhook) GetInfo() *plugin.Info {
+	elements := []*plugin.ConfigOption{
+		{
+			Name: "method",
+			Type: "option",
+			Label: map[string]string{
+				"en_US": "HTTP Method",
+				"de_DE": "HTTP-Methode",
+			},
+			Help: map[string]string{
+				"en_US": "HTTP request method used for the web request.",
+				"de_DE": "HTTP-Methode für die Anfrage.",
+			},
+			Default:  "POST",
+			Required: true,
+			Options: map[string]string{
+				// https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods
+				"GET":     "GET",
+				"HEAD":    "HEAD",
+				"POST":    "POST",
+				"PUT":     "PUT",
+				"DELETE":  "DELETE",
+				"CONNECT": "CONNECT",
+				"OPTIONS": "OPTIONS",
+				"TRACE":   "TRACE",
+				"PATCH":   "PATCH",
+			},
+		},
+		{
+			Name: "url_template",
+			Type: "string",
+			Label: map[string]string{
+				"en_US": "URL Template",
+				"de_DE": "URL-Template",
+			},
+			Help: map[string]string{
+				"en_US": "URL, optionally as a Go template over the current plugin.NotificationRequest.",
+				"de_DE": "URL, optional als Go-Template über das zu verarbeitende plugin.NotificationRequest.",
+			},
+			Required: true,
+		},
+		{
+			Name: "request_body_template",
+			Type: "string",
+			Label: map[string]string{
+				"en_US": "Request Body Template",
+				"de_DE": "Anfragedaten-Template",
+			},
+			Help: map[string]string{
+				"en_US": "Go template applied to the current plugin.NotificationRequest to create an request body.",
+				"de_DE": "Go-Template über das zu verarbeitende plugin.NotificationRequest zum Erzeugen der mitgesendeten Anfragedaten.",
+			},
+			// Default as a minimal working example to send a JSON object.
+			Default: `{"id": {{.Incident.Id}}, "type": "{{.Event.Type}}", "name": "{{.Object.Name}}", "severity": "{{.Incident.Severity}}", "url": "{{.Incident.Url}}"}`,
+		},
+	}
+
+	configAttrs, err := json.Marshal(elements)
+	if err != nil {
+		panic(err)
+	}
+
+	return &plugin.Info{
+		Name:             "Webhook",
+		Version:          internal.Version.Version,
+		Author:           "Icinga GmbH",
+		ConfigAttributes: configAttrs,
+	}
+}
+
+func (ch *Webhook) SetConfig(jsonStr json.RawMessage) error {
+	err := json.Unmarshal(jsonStr, ch)
+	if err != nil {
+		return err
+	}
+
+	tmplFuncs := template.FuncMap{
+		"json": func(a any) string {
+			var buff bytes.Buffer
+			if err := json.NewEncoder(&buff).Encode(a); err != nil {
+				panic(err)
```

Comment:
[`FuncMap`](https://pkg.go.dev/text/template#FuncMap) explicitly allows to return errors:

> or two return values of which the second has type error. In that case, if the second (error) return value evaluates to non-nil during execution, execution terminates and Execute returns that error.

So there seems to be no need to panic here.